lib/cdo/share_filtering.rb

```diff
@@ -0,0 +1,40 @@
+require 'cdo/regexp'
+require 'cdo/geocoder'
+require 'cdo/web_purify'
+
+USER_ENTERED_TEXT_INDICATORS = ['TITLE', 'TEXT', 'title name\=\"VAL\"']
+ShareFailure = Struct.new(:type, :content)
+
+# Utilities for finding personally-identifiable and profane content in user
+# submitted programs.
+module ShareFiltering
+  module FailureType
+    EMAIL = 'email'
+    ADDRESS = 'address'
+    PHONE = 'phone'
+    PROFANITY = 'profanity'
+  end
```

Comment:
Hmm, here was my thinking on using module constants over hash keys/values—on the IDE side, this enables "find usages" and completion for the constants, and since we're never e.g. iterating through the keys/values, there are only a few known types, and it is never modified programmatically it might not be the best use case for a hash. The intent here is to define constants for the different the known failure types that consumers can use to decide which error messages to output based on the found failure.

As a hash, consuming the constants might look something like:

![ide-failures-ux](https://cloud.githubusercontent.com/assets/206973/16695023/49aa0cb8-44f2-11e6-985b-5f6ab1b5d900.gif)

As module constants, you can auto-complete / find usages like:

![ide-failures-module-ux](https://cloud.githubusercontent.com/assets/206973/16695033/5221da06-44f2-11e6-9c86-233cd3dd34bd.gif)